File: login-server/src/main/java/work/fking/pangya/login/net/ClientPacketHandler.java
package work.fking.pangya.login.net;

import io.netty.buffer.ByteBuf;
import work.fking.pangya.login.LoginServer;
import work.fking.pangya.login.Player;

public interface ClientPacketHandler {

    void handle(LoginServer server, Player player, ByteBuf packet);
}

File: login-server/src/main/kotlin/work/fking/pangya/login/net/ClientPacketHandler.kt
package work.fking.pangya.login.net

import io.netty.buffer.ByteBuf
import work.fking.pangya.login.LoginServer
import work.fking.pangya.login.Player

fun interface ClientPacketHandler {
    fun handle(server: LoginServer, player: Player, packet: ByteBuf)
}
